packages/spiderfier/directives/marker-spider.ts

```diff
@@ -0,0 +1,227 @@
+import {
+  Directive,
+  EventEmitter,
+  Input,
+  NgZone,
+  OnChanges,
+  OnDestroy,
+  OnInit,
+  Output,
+  SimpleChange
+} from '@angular/core';
+
+import {SpiderManager} from '../services/managers/spider-manager';
+import {InfoWindowManager, MarkerManager} from '@agm/core';
+import {Marker} from '../../core/services/google-maps-types';
+
+import {LegColorOptions, SpiderOptions} from '../services/google-spider-types';
+
+/**
+ * AgmMarkerSpider spiderfies map marker if they are near together
+ *
+ * ### Example
+ * ```typescript
+ * import { Component } from '@angular/core';
+ *
+ * @Component({
+ *  selector: 'my-map-cmp',
+ *  styles: [`
+ *    agm-map {
+ *      height: 300px;
+ *    }
+ * `],
+ *  template: `
+ *    <agm-map [latitude]="lat" [longitude]="lng" [zoom]="zoom">
+ *      <agm-marker-spider>
+ *        <agm-marker [latitude]="lat" [longitude]="lng" [label]="'M'">
+ *        </agm-marker>
+ *        <agm-marker [latitude]="lat2" [longitude]="lng2" [label]="'N'">
+ *        </agm-marker>
+ *      </agm-marker-spider>
+ *    </agm-map>
+ *  `
+ * })
+ * ```
+ */
+@Directive({
+  selector: 'agm-marker-spider',
+  providers: [
```

Comment:
as want to start implementing ChangeDetectionStrategy.OnPush for all component: can you add it here and test that it runs with OnPush?

Reply:
The changeDetection attribute is only available for components, not directives. Please update the `@agm/core` package and I'll test if this gives any problems in my package